### ISSUE TITLE
OnsenUI 2 quickstart "npm install" errors

### ISSUE THREAD

**Issue by jpdelmundo** [OnsenUI/OnsenUI#1275]
**OS: Windows 7
Additional: Python wasn't mentioned as a requirement so I got the error first and installed Python. Then after running "npm install" again, here's the result...**

npm WARN deprecated graceful-fs@3.0.8: graceful-fs version 3 and before will fail on newer node releases. Please update to graceful-fs@^4.0.0 as soon as possible.
npm WARN optional dep failed, continuing fsevents@1.0.8
npm WARN deprecated lodash@1.0.2: lodash@<3.0.0 is no longer maintained. Upgrade to lodash@^4.0.0.
npm WARN deprecated graceful-fs@1.2.3: graceful-fs version 3 and before will fail on newer node releases. Please update to graceful-fs@^4.0.0 as soon as possible.

> bufferutil@1.2.1 install F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\bufferutil
> node-gyp rebuild

F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\bufferutil>if not defined npm_config_node_gyp (node "C:\Program Files (x86)\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\
bin\node-gyp.js" rebuild )  else (node  rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppBuild.targets(366,5): warning MSB8003: Could not find WindowsSDKDir variable from the registry.  TargetFrameworkVersion or PlatformToolset may be set to an invalid version number. [F:\WORK\Projects\myapp\apps\myapp-onsenu
i2\node_modules\browser-sync\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\bufferutil\build\bufferutil.vcxproj]
TRACKER : error TRK0005: Failed to locate: "CL.exe". The system cannot find the file specified. [F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\bufferutil\build\bufferutil.vcxproj]

gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\bufferutil
gyp ERR! node -v v4.2.6
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok

> utf-8-validate@1.2.1 install F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\utf-8-validate
> node-gyp rebuild

F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\utf-8-validate>if not defined npm_config_node_gyp (node "C:\Program Files (x86)\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-
gyp\bin\node-gyp.js" rebuild )  else (node  rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppBuild.targets(366,5): warning MSB8003: Could not find WindowsSDKDir variable from the registry.  TargetFrameworkVersion or PlatformToolset may be set to an invalid version number. [F:\WORK\Projects\myapp\apps\myapp-onsenu
i2\node_modules\browser-sync\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\utf-8-validate\build\validation.vcxproj]
TRACKER : error TRK0005: Failed to locate: "CL.exe". The system cannot find the file specified. [F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\utf-8-validate\build\validation.vcxproj]

gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\utf-8-validate
gyp ERR! node -v v4.2.6
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok
npm WARN optional dep failed, continuing bufferutil@1.2.1

> utf-8-validate@1.2.1 install F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\utf-8-validate
> node-gyp rebuild

F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\utf-8-validate>if not defined npm_config_node_gyp (node "C:\Program Files (x86)\nodejs\node_modules\npm\bin\n
ode-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppBuild.targets(366,5): warning MSB8003: Could not find WindowsSDKDir variable from the registry.  TargetFrameworkVersion or PlatformToolset may be set to an invalid version number. [F:\WORK\Projects\myapp\apps\myapp-onsenu
i2\node_modules\browser-sync\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\utf-8-validate\build\validation.vcxproj]
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppBuild.targets(403,5): error MSB3191: Unable to create directory "Release\obj\validation\validation.tlog\". The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the
 directory name must be less than 248 characters. [F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\utf-8-validate\build\validation.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\utf-8-validate
gyp ERR! node -v v4.2.6
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok
npm WARN optional dep failed, continuing utf-8-validate@1.2.1

> bufferutil@1.2.1 install F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\bufferutil
> node-gyp rebuild

F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\bufferutil>if not defined npm_config_node_gyp (node "C:\Program Files (x86)\nodejs\node_modules\npm\bin\node-
gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppBuild.targets(366,5): warning MSB8003: Could not find WindowsSDKDir variable from the registry.  TargetFrameworkVersion or PlatformToolset may be set to an invalid version number. [F:\WORK\Projects\myapp\apps\myapp-onsenu
i2\node_modules\browser-sync\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\bufferutil\build\bufferutil.vcxproj]
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppBuild.targets(312,5): error MSB3491: Could not write lines to file "Release\obj\bufferutil\bufferutil.tlog\bufferutil.lastbuildstate". The specified path, file name, or both are too long. The fully qualified file name must be less
than 260 characters, and the directory name must be less than 248 characters. [F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\bufferutil\build\bufferutil.v
cxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd F:\WORK\Projects\myapp\apps\myapp-onsenui2\node_modules\browser-sync\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\bufferutil
gyp ERR! node -v v4.2.6
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok
npm WARN optional dep failed, continuing utf-8-validate@1.2.1
npm WARN optional dep failed, continuing bufferutil@1.2.1
rimraf@2.5.2 node_modules\rimraf
└── glob@7.0.0 (path-is-absolute@1.0.0, inherits@2.0.1, inflight@1.0.4, once@1.3.3, minimatch@3.0.0)

gulp-serve@1.2.0 node_modules\gulp-serve
├── serve-static@1.10.2 (escape-html@1.0.3, parseurl@1.3.1, send@0.13.1)
├── connect@3.4.1 (parseurl@1.3.1, utils-merge@1.0.0, debug@2.2.0, finalhandler@0.4.1)
└── gulp-util@3.0.7 (array-differ@1.0.0, array-uniq@1.0.2, beeper@1.1.0, object-assign@3.0.0, lodash._reevaluate@3.0.0, lodash._reescape@3.0.0, lodash._reinterpolate@3.0.0, replace-ext@0.0.1, minimist@1.2.0, fancy-log@1.2.0, has-gulplog@0.1.0, vinyl@0.5.3, gulplog@1.0.0, lodash.template@3.6.2, chalk
@1.1.1, through2@2.0.1, multipipe@0.1.2, dateformat@1.0.12)

gulp@3.9.1 node_modules\gulp
├── interpret@1.0.0
├── pretty-hrtime@1.0.2
├── deprecated@0.0.1
├── archy@1.0.0
├── minimist@1.2.0
├── tildify@1.1.2 (os-homedir@1.0.1)
├── semver@4.3.6
├── v8flags@2.0.11 (user-home@1.1.1)
├── chalk@1.1.1 (escape-string-regexp@1.0.5, supports-color@2.0.0, has-ansi@2.0.0, strip-ansi@3.0.1, ansi-styles@2.2.0)
├── orchestrator@0.3.7 (stream-consume@0.1.0, sequencify@0.0.7, end-of-stream@0.1.5)
├── liftoff@2.2.0 (extend@2.0.1, rechoir@0.6.2, flagged-respawn@0.3.1, findup-sync@0.3.0, resolve@1.1.7)
├── vinyl-fs@0.3.14 (graceful-fs@3.0.8, strip-bom@1.0.0, defaults@1.0.3, vinyl@0.4.6, through2@0.6.5, mkdirp@0.5.1, glob-stream@3.1.18, glob-watcher@0.0.6)
└── gulp-util@3.0.7 (array-differ@1.0.0, array-uniq@1.0.2, object-assign@3.0.0, lodash._reescape@3.0.0, lodash._reevaluate@3.0.0, beeper@1.1.0, lodash._reinterpolate@3.0.0, replace-ext@0.0.1, has-gulplog@0.1.0, fancy-log@1.2.0, vinyl@0.5.3, lodash.template@3.6.2, gulplog@1.0.0, through2@2.0.1, multi
pipe@0.1.2, dateformat@1.0.12)

browser-sync@2.11.1 node_modules\browser-sync
├── ucfirst@1.0.0
├── async-each-series@0.1.1
├── longest@1.0.1
├── emitter-steward@1.0.0
├── easy-extender@2.3.2
├── dev-ip@1.0.1
├── immutable@3.7.6
├── ua-parser-js@0.7.10
├── opn@3.0.3 (object-assign@4.0.1)
├── query-string@2.4.2 (strict-uri-encode@1.1.0)
├── browser-sync-client@2.4.1 (fresh@0.3.0, etag@1.7.0)
├── portscanner@1.0.0 (async@0.1.15)
├── connect@3.4.1 (utils-merge@1.0.0, parseurl@1.3.1, debug@2.2.0, finalhandler@0.4.1)
├── serve-static@1.10.2 (escape-html@1.0.3, parseurl@1.3.1, send@0.13.1)
├── resp-modifier@5.0.2 (debug@2.2.0, minimatch@2.0.10)
├── fs-extra@0.26.5 (path-is-absolute@1.0.0, klaw@1.1.3, jsonfile@2.2.3, graceful-fs@4.1.3)
├── eazy-logger@2.1.2 (opt-merger@1.1.1, tfunk@3.0.1)
├── foxy@11.1.4 (http-proxy@1.13.2, lodash.merge@3.3.2, resp-modifier@4.0.4)
├── meow@3.3.0 (object-assign@3.0.0, minimist@1.2.0, camelcase-keys@1.0.0, indent-string@1.2.2)
├── chokidar@1.4.1 (path-is-absolute@1.0.0, inherits@2.0.1, glob-parent@2.0.0, async-each@0.1.6, is-glob@2.0.1, is-binary-path@1.0.1, anymatch@1.3.0, readdirp@2.0.0)
├── micromatch@2.3.5 (filename-regex@2.0.0, normalize-path@2.0.1, is-glob@2.0.1, is-extglob@1.0.0, lazy-cache@0.2.7, extglob@0.3.2, expand-brackets@0.1.4, array-unique@0.2.1, regex-cache@0.4.2, arr-diff@2.0.0, kind-of@3.0.2, parse-glob@3.0.4, object.omit@2.0.0, braces@1.8.3)
├── serve-index@1.7.3 (escape-html@1.0.3, parseurl@1.3.1, batch@0.5.3, mime-types@2.1.10, http-errors@1.3.1, accepts@1.2.13, debug@2.2.0)
├── bs-recipes@1.1.0
├── localtunnel@1.8.1 (openurl@1.1.0, debug@2.2.0, yargs@3.29.0, request@2.65.0)
├── lodash@3.10.1
├── browser-sync-ui@0.5.18 (connect-history-api-fallback@1.1.0, stream-throttle@0.1.3, weinre@2.0.0-pre-I0Z7U9OV)
└── socket.io@1.3.7 (has-binary-data@0.1.3, debug@2.1.0, socket.io-parser@2.2.4, socket.io-adapter@0.3.1, engine.io@1.5.4, socket.io-client@1.3.7)

gulp-bower@0.0.10 node_modules\gulp-bower
├── walk@2.3.3 (foreachasync@3.0.0)
├── through2@0.6.2 (xtend@4.0.1, readable-stream@1.0.33)
├── gulp-util@3.0.1 (lodash._reinterpolate@2.4.1, minimist@1.2.0, lodash@2.4.2, vinyl@0.4.6, chalk@0.5.1, lodash.template@2.4.1, multipipe@0.1.2, dateformat@1.0.12)

**Comment by IliaSky** [comment]
Sorry for the late answer - I guess your issue doesn't look very appealing to the development team.
Github has a new feature for issue templates, but I guess we haven't made ours yet.
I am going to separate the the answer into a few points:

1. When you paste a log on github please use triple tick quotes \`\`\` on both sides, so that it can be seen better as a log instead of just text. Also try to avoid putting 150+ line logs - this one isn't even full, so I'm guessing you either failed to copy everything or github just trimmed it down for you, because it's too long.
2. Instead of the whole log you could just include the error itself. Also when the error is inside the resullt of `npm install` probably that means that it's a bug of one of the dependencies, not Onsen UI. So you should report the bug over there. It's also likely that if you google the exact error you will find a stackoverflow question or github issue already referencing that error and maybe the solution for it as well.
3. In your case the first thing to see is `TRACKER : error TRK0005: Failed to locate: "CL.exe".` - the [first result](http://stackoverflow.com/questions/33183161/node-gyp-error-tracker-error-trk0005-failed-to-locate-cl-exe-the-system-c) of a google search for this has an accepted answer as well as a comment pointing to the [node-gyp installation docs on github](https://github.com/nodejs/node-gyp#user-content-installation). - there you can also see the `python 2.7` requirement. So make sure you didn't accidentally install `python 3` instead of `2.7`, but I guess maybe in that case you would have gotten a different error.
4. From what I can see the next errors all say that some path is too long and they cannot write to a file with a path larger than 260 characters, but all of them later result in `npm WARN optional dep failed`, so not that important since it's an optional dependency.
5. I guess an error on our part is the fact that we're using so many things in a simple demo. I guess for most developers doing `npm install && gulp serve` seems easier, but for ones who don't care about dependencies and just want to try out onsen none of this is even needed. Onsen UI is a frontend ui lib and thus the only thing you need to do to use it is just include it in a script tag. And actually it's over there already. The whole `npm install` is just to give you a server and some extras like browser refresh etc. You don't even need it. If you have python already installed you can just do `python -m SimpleHTTPServer 8000` in the `www` folder and open `localhost:8000` in the browser. Alternatively you could even just open the `index.html` from your file explorer, but I usually it's better if you have a server for things like this.

To summarise:
This is an issue with the dependencies for the [onsenui2-quickstart](https://github.com/OnsenUI/onsenui2-quickstart) repo and actually in it's dependencies, not the product, so the issue should be at the repo of the specific dependency, and only if you don't find an answer to your problem over there already.
If you don't care about the server and additional features just run `python -m SimpleHTTPServer 8000` instead.

**Comment by anujchaudhary1486** [comment]
@ IliaSky

**Comment by IliaSky** [comment]
@anujchaudhary1486
In the [node-gyp docs](https://github.com/nodejs/node-gyp#user-content-installation) they say that 3.x is not supported, so if you were using that then it may cause a problem.

Also the problem is caused dependencies of the demo itself, not by onsen ui, so this issue should be in that repo, but nevermind. The demo provides some additional features such as browser refresh etc, which are not really related to onsen ui and which seem to be the ones failing the `npm install` command.

Anyway I'm glad that my answer in this issue was able to help you solve the problem.

I guess in the end even though python 3 has been out for quite a while now people still use 2.7 a lot...
